### PR TITLE
Agent can just "use" an already created EK

### PR DESCRIPTION
**Pull request by maugustosilva**
This commit is a first attempt at giving keylime de ability to use an
    already generated Endorsement Key (EK) on a TPM. The use case and
    rationale for this is well described on #313.

    A new option is introduced under `[cloud_agent]`, name `ek_handle`. When
    the default value of `generate` (wanted to keep the familiar term
    already present on other options, such as `tls_dir`) is found, the
    normal agent startup process, which includes clearing/start, ownership,
    flush/evict and EK creation, will be executed. If `ek_handle` points to
    an actual value (e.g., `0x81000000`), then all the initial steps
    previously described will be skipped and the agent will just start from
    the "create AIK" step. In the latter scenario, the agent will make sure
    that particular handle is not flushed on exit.

    Resolves: #313

### CODE REVIEW

**Comment by lukehinds** [comment]
@jetwhiz could you take a look too, see #313 for context.

**Comment by lukehinds** [comment]
how could I try this out @maugustosilva  ?

**Comment by maugustosilva** [comment]
apologies for the long delay answering @lukehinds. Here is how I've been testing..

Step 1: Fully boot a node, and then issue the following commands against a "clean" (v)TPM:

```
tpm2_startup --version
tpm2_startup -c
tpm2_changeauth -c o temp4now
tpm2_changeauth -c e temp4now
tpm2_changeauth -c o -p temp4now temp4now
tpm2_changeauth -c e -p temp4now temp4now
tpm2_getcap handles-persistent
tpm2_createek -c - -G rsa -u /tmp/test -p rsxBpUzLpR28mHPlC5NN -w temp4now -P temp4now
tpm2_getcap handles-persistent
```

Step 2: get the handle from the last command in step 1 (e.g., `0x81000000`), and add it under the section `[cloud_agent`] in `/etc/keylime.conf`

```
...
[cloud_agent]
...
ek_handle = 0x81000000
...
```

Step3: start `keylime_agent`

**Comment by lukehinds** [comment]
Nice! 

If you can, it would be nice to have a section over at keylime/keylime-docs , something I guess like "Agent setup for nodes with existing EK"

**Comment by maugustosilva** [comment]
@lukehinds Documentation PR submitted: https://github.com/keylime/keylime-docs/pull/51

File: keylime.conf [hunk 1]
```diff
@@ -103,6 +103,11 @@ tpm_hash_alg = sha256
 tpm_encryption_alg = rsa
 tpm_signing_alg = rsassa
 
+# If an EK was already created on the TPM (e.g., with "tpm2_createek") and
+# keylime should just use it, just change from "generate" to the actual
```
Comment (lukehinds) [inline keylime.conf:107]:
```
# If an EK is already present on the TPM (e.g., with "tpm2_createek") and
# you require keylime to use this EK, change "generate" to the actual EK 
# handle (e.g. "0x81000000"). The keylime agent will then not attempt to 
# create a new EK upon startup, and neither will it flush the EK upon exit
ek_handle = generate
```

> Hope the above changes are ok with you.

**Pull request merged**